File: gastoclaro_app/lib/models/fixed_expense.dart
class FixedExpense {
  final int id;
  final String name;
  final String? category;
  final double amount;
  final String currency;
  final int? dueDay;
  final String frequency;
  final bool isMandatory;
  final bool isActive;
  final String? notes;

  FixedExpense({
    required this.id,
    required this.name,
    required this.category,
    required this.amount,
    required this.currency,
    required this.dueDay,
    required this.frequency,
    required this.isMandatory,
    required this.isActive,
    required this.notes,
  });

  factory FixedExpense.fromJson(Map<String, dynamic> json) {
    double toDouble(dynamic value) {
      if (value is int) {
        return value.toDouble();
      }

      if (value is double) {
        return value;
      }

      return double.tryParse(value.toString()) ?? 0;
    }

    int? toNullableInt(dynamic value) {
      if (value == null) {
        return null;
      }

      if (value is int) {
        return value;
      }

      return int.tryParse(value.toString());
    }

    return FixedExpense(
      id: toNullableInt(json['id']) ?? 0,
      name: json['name']?.toString() ?? '',
      category: json['category']?.toString(),
      amount: toDouble(json['amount']),
      currency: json['currency']?.toString() ?? 'PEN',
      dueDay: toNullableInt(json['due_day']),
      frequency: json['frequency']?.toString() ?? 'monthly',
      isMandatory: json['is_mandatory'] == true,
      isActive: json['is_active'] == true,
      notes: json['notes']?.toString(),
    );
  }
}

File: gastoclaro-api/app/Models/FixedExpense.php
<?php

declare(strict_types=1);

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class FixedExpense extends Model
{
    protected $fillable = [
        'user_id',
        'name',
        'category',
        'amount',
        'currency',
        'due_day',
        'frequency',
        'is_mandatory',
        'is_active',
        'notes',
    ];

    protected function casts(): array
    {
        return [
            'amount' => 'decimal:2',
            'due_day' => 'integer',
            'is_mandatory' => 'boolean',
            'is_active' => 'boolean',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}
